### ISSUE TITLE
ASM Installation fails with Permission Denied

### ISSUE THREAD

**Issue by kunallimaye** [terraform-google-modules/terraform-google-kubernetes-engine#884]
Use of [`asm` module](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/v14.2.0/modules/asm), fails with the following error message:
```bash
Running ./install_asm --verbose --project_id kubernetes-project-id --cluster_name frontend-k8s-cluster --cluster_location us-central1-b --mode install --enable_cluster_labels --enable_cluster_roles --managed
./install_asm: line 1: syntax error near unexpected token `<'
./install_asm: line 1: `<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object.</Details></Error>'
```

**Comment by kunallimaye** [comment]
Line #32 of .terraform/modules/asm/modules/asm/scripts/install_asm.sh should be updated:  

`- curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm`
`+ curl https://raw.githubusercontent.com/GoogleCloudPlatform/anthos-service-mesh-packages/staging-"${ASM_VERSION}"/scripts/asm-installer/install_asm > install_asm`

**Comment by bharathkkb** [comment]
Hi @kunallimaye 
Thanks for the report. That seems odd, are you overriding the version? I tried a quick check with 1.8 in a container without auth and that works. 
```
docker run --rm curlimages/curl:7.76.1 https://storage.googleapis.com/csm-artifacts/asm/install_asm_1.8
```
If I specify a non existent version, I do get this error
```
 docker run --rm curlimages/curl:7.76.1 https://storage.googleapis.com/csm-artifacts/asm/install_asm_1.99
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   223  100   223    0     0   2563      0 --:--:-- --:--:-- --:--:--  2563
<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to the Google Cloud Storage obj
```

**Comment by kunallimaye** [comment]
You are correct - I was using the wrong version for asm `istio-1.9.3-asm.2`.
It works fine once I set that value to `1.9`.

**Comment by giulianobr** [comment]
Still not working with ASM 1.10.2

**Comment by giulianobr** [comment]
To fix, just write 1.10, removing the minor version number.
